### DISCUSSION TITLE
Using batchrunner to loop through multiple netCDF files

### DISCUSSION THREAD

**Discussion by NeKcenos** [mesa/mesa#1064] [Q&A]
Hello. I am new to python and the mesa project. I am developing a mesa model to simulate future distribution patterns, for a particular shark species, in relation to changing ocean temperatures as a result of climate change.

My main dataset I am using for this is a bunch of netCDF4 files for sea surface temperature for each of the 4 climatic projections (ie. RCP 2.6, 4.5, 6 and RCP 8.5).

I realise I can use batchrunner to automate the process of continuous running for the model (lets say I want to run the model 50 times over), but I also want the model to iterate through each netCDF projection file as well. So for example, 50 times for RCP 2.6, then run 50 more times for RCP 4.5 and so on until it reaches the last netCDF file.

Is there a way this can be achieved? If so, could someone give me or direct me to some example code that I can test around with?
Thanks.

**Comment by tpike3** [comment]
@NeKcenos BatchRunner can do this. You want to set your RCPs as variable parameter and then set iterations to 50. As a generic for instance

```
variable_params = {"RCP": [<RCP2.6 object>, <RCP4.5 Object>, <RCP6 Object>, <RCP8 Object>] }

batch_run = BatchRunner(SharkModel,
                        variable_params,
                        iterations=50,
                        max_steps=1000,
                        )
batch_run.run_all()
```

You can also use BatchRunnerMP where you put each model on a different core to reduce overall run time.
